### PR TITLE
Module 2 Merge

### PR DESCRIPTION
**Pull request by rawsashimi1604**
# Description

Added notification feature
Added device logging feature

Fixes # (issue)

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Manual testing of Notification feature e2e
- [x] Manual testing of device logging e2e

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      },
+
+  }
+};
+
+
+
+function CarbonChart() {
+  return (
+    <Container maxWidth={1600} mt={2} centerContent>
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js:88]: @rawsashimi1604 remove maxwidth and center

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  Tab,
+  TabPanel,
+  Box,
+  Center,
+  Container,
+  Text,
+  Heading
+} from '@chakra-ui/react';
+import { Line } from "react-chartjs-2";
+import { getByPlaceholderText } from '@testing-library/react';
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js:16]: @tryhardlaijun this is for?

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/EfficiencyBar.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            backgroundColor: 'rgba(255, 99, 132, 0.5)',
+        },
+    ],
+};
+
+export function EfficiencyBar() {
+
+    return (
+
+        <Container maxWidth={1600} mt={2} centerContent>
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/Analytics/EfficiencyBar.js:65]: @rawsashimi1604 remove maxwidth and center content

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/UsageBar.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          borderColor: "rgb(255, 99, 132)",
+          data: [27, 10, 5, 2, 20, 30, 45],
+        },
+      ],
+    };
+function UsageBar() {
+    
+
+    return (
+      <Container maxWidth={1600} mt={2} centerContent>
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/Analytics/UsageBar.js:33]: @rawsashimi1604 remove maxwidth and centercontent

File: SmartHomeManager/SmartHomeManager.View/src/components/DeviceLogging.js [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    BarElement,
+    Title,
+    Tooltip,
+    Legend
+);
+
+/*
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/DeviceLogging.js:24]: @rawsashimi1604 if not needed anymore, delete the comments

File: SmartHomeManager/SmartHomeManager.View/src/components/DeviceLogging.js [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+        const value = event.target.value;
+        setDevice1ID(value);
+    };
+
+    const fetchWeeklyLog = async (id, date) => {
+        try {
+            //create the fetch request
+            const res = await axios.get(`https://localhost:7140/api/DeviceLog/${id}/${date}`)
+
+            console.log(res)
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/DeviceLogging.js:195]: @rawsashimi1604 del logs

File: SmartHomeManager/SmartHomeManager.View/src/components/Notification/Notification.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+    const SESSION_ACCOUNT_GUID = "9D2B0228-4D0D-4C23-8B49-01A698857709";
+    const {isOpen, onClose, onToggle} = useDisclosure();
+    const [notifications, setNotifications] = useState([]);
+    const [errors, setErrors] = useState(null);
+
+    function getNotifications() {
+        NotificationService.getNotificationsByAccountId(SESSION_ACCOUNT_GUID)
+            .then(response => {
+                console.log(response.data)
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/Notification/Notification.js:40]: @rawsashimi1604 remove logs

File: SmartHomeManager/SmartHomeManager.View/src/components/Notification/TestNotification.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+      // handling the post request...
+      NotificationService.sendNotification(
+        {
+          "notificationObject": {
+            "message": input,
+            "accountId": SESSION_ACCOUNT_GUID
+          }
+        }
+      ).then(response => {
+        console.log(response)
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/Notification/TestNotification.js:33]: @rawsashimi1604 remove logs

File: SmartHomeManager/SmartHomeManager.View/src/components/Notification/TestNotificationModal.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import React, {useState} from "react";
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/components/Notification/TestNotificationModal.js:1]: @rawsashimi1604 if this is a test can delete?

File: SmartHomeManager/SmartHomeManager.View/src/index.js [hunk 10]
```diff
@@ -6,11 +6,15 @@ import reportWebVitals from './reportWebVitals';
 import { ChakraProvider } from '@chakra-ui/react';
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/index.js:6]: @rawsashimi1604 🤔🤔🤔🤔🤔🤔🤔🤔

File: SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    Box
+} from '@chakra-ui/react'
+
+export default function DeviceConfig() {
+
+    const [DropDownValue, setDropDownValue] = useState(null);
+    const [sliderValue, setSliderValue] = useState(50)
+
+
+    const labelStyles = {
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js:18]: @rawsashimi1604 use chakra prop styles

File: SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+        mt: '2',
+        ml: '-2.5',
+        fontSize: 'sm',
+    }
+
+    const dropDownHandler = (e) => {
+        let value = e.target.value;
+        setDropDownValue(value);
+        if (value === "option1") {
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js:27]: @rawsashimi1604 be careful, would just be better to set as value === 50/30 instead

File: SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+            setSliderValue(30)
+        }
+    };
+
+    return (
+        <div> <strong>Set Threshold Limit for devices</strong>
+            <br></br>
+            <br></br>
+            <Select placeholder='Select Device'
+                width={"200px"}
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js:39]: @rawsashimi1604 why so specific?

File: SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+                onChange={(e) => dropDownHandler(e)}>
+                <option value='option1'>Device 1</option>
+                <option value='option2'>Device 2</option>
+            </Select>
+            {DropDownValue ? (
+                <div style={{ width: "600px" }}>
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js:45]: @rawsashimi1604 use chakra prop styles

File: SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+                    <div><br></br>Limit is: <strong>{sliderValue}</strong></div>
+                </div>
+            ) : (
+                <div>
+                    <strong>Please select any value from dropdown</strong>
+                </div>
+            )}
+        </div>
+        );
+}
```
Comment (fabianchua6) [inline SmartHomeManager/SmartHomeManager.View/src/pages/DeviceConfig.js:87]: @rawsashimi1604 avoid using divs

**Review by fabianchua6 — COMMENTED** [review]
@rawsashimi1604 left comments accordingly on frontend, please assign respective members to fix ASAP

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js (hunk 1 above)
Comment (rawsashimi1604) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js:88]: @junweilam @jianweiiii please fix under branch `fix/analysis-front-end`, submit PR when ready

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js (hunk 2 above)
Comment (rawsashimi1604) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js:16]: @junweilam @jianweiiii please fix under branch `fix/analysis-front-end`, submit PR when ready

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/EfficiencyBar.js (hunk 3 above)
Comment (rawsashimi1604) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/EfficiencyBar.js:65]: @junweilam @jianweiiii please fix under branch `fix/analysis-front-end`, submit PR when ready

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/UsageBar.js (hunk 4 above)
Comment (rawsashimi1604) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/UsageBar.js:33]: @junweilam @jianweiiii please fix under branch `fix/analysis-front-end`, submit PR when ready

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js (hunk 1 above)
Comment (junweilam) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js:88]: @rawsashimi1604 fixed

Comment (rawsashimi1604) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/CarbonChart.js:88]: @junweilam ok please submit a pull request to our module once all the changes are done

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/EfficiencyBar.js (hunk 3 above)
Comment (junweilam) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/EfficiencyBar.js:65]: @rawsashimi1604 fixed

File: SmartHomeManager/SmartHomeManager.View/src/components/Analytics/UsageBar.js (hunk 4 above)
Comment (junweilam) [reply SmartHomeManager/SmartHomeManager.View/src/components/Analytics/UsageBar.js:33]: @rawsashimi1604 fixed

**Pull request merged**